torch/_inductor/config.py

```diff
@@ -193,3 +193,92 @@ class trace:
     # Upload the .tar.gz file
     # Needs to be overriden based on specific environment needs
     upload_tar = None
+
+
+class InductorConfigContext:
+    static_memory: bool
+    matmul_tune: str
+    matmul_padding: bool
+    triton_autotune: bool
+    triton_bmm: bool
+    triton_mm: str
+    triton_convolution: str
+    rematerialize_threshold: int
+    rematerialize_acc_threshold: int
+
+    def _save(self):
+        self.static_memory = triton.cudagraphs
+        self.matmul_tune = triton.mm
+        self.matmul_padding = shape_padding
+        self.triton_autotune = triton.autotune
+        self.triton_bmm = triton.use_bmm
+        self.triton_mm = triton.mm
+        self.triton_convolution = triton.convolution
+        self.rematerialize_threshold = realize_reads_threshold
+        self.rematerialize_acc_threshold = realize_acc_reads_threshold
+
+    def _apply(self):
+        triton.cudagraphs = self.static_memory
+        triton.mm = self.matmul_tune
+        shape_padding = self.matmul_padding
+        triton.autotune = self.triton_autotune
+        triton.use_bmm = self.triton_bmm
+        triton.mm = self.triton_mm
+        triton.convolution = self.triton_convolution
+        realize_reads_threshold = self.rematerialize_threshold
+        realize_acc_reads_threshold = self.rematerialize_acc_threshold
```

Comment:
What would happen if I invoke `torch.compile` from two threads with different configurations at the same time?

Reply:
It would take one or the other setting, but this is just a preliminary implementation...